src/button/Button.js

```diff
@@ -72,14 +88,31 @@ export default class Button extends React.Component {
 
     const buttonLogicProps = pick(this.props, ['baseState', 'buttonState', 'onClick', 'stableSuccess', 'timerMillis']);
 
+    const getButtonType = () => {
+      const { type, primary, flat } = this.props;
+      return type || (primary && 'primary') || (flat && 'flat') || 'default';
+    };
+
+    const wrapperStyle = {
+      display: fluid ? 'block' : 'inline-block',
+      width: fluid ? '100%' : null
+    };
+
+    const isIconButton = () => this.props.icon && !this.props.label;
+
+    const className = cx(
```

Comment:
I'm trying to switch to this convention for the state classNames:
```js
`is-${state}`  // is-default, is-icon-button, is-...
```

I added a small util in the last PR to make it easier: https://github.com/buildo/react-components/blob/master/src/utils/index.js#L6